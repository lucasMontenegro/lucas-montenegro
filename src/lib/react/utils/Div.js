import React from "react"
import PropTypes from "prop-types"
function Div ({ box, color, style, ...props }, ref) {
  return (
    <div
      ref={ref}
      style={{
        padding: 4,
        border: color && `2px solid ${color}`,
        margin: 4,
        minHeight: box,
        ...style,
      }}
      {...props}
    />
  )
}
export default React.forwardRef(Div)
Div.propTypes = {
  box: PropTypes.number,
  color: PropTypes.string,
  style: PropTypes.object,
}