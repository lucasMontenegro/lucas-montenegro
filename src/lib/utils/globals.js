import g from "global"
// renaming "global" to "globals"
// declaring a variable named "global" may cause trouble
export default g