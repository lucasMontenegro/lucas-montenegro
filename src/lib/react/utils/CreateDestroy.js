import React, { useState } from "react"
import Div from "lib/react/utils/Div"
import Button from "@material-ui/core/Button"
export default function CreateDestroy ({ id, box, Component, props }) {
  const [render, setRender] = useState(false)
  return (
    <Div box={box} color="blue">
      <Div>
        <Button
          id={id}
          variant="outlined"
          color="primary"
          onClick={() => setRender(render => !render)}
        >
          CREATE/DESTROY
        </Button>
      </Div>
      <Div><em>{render ? `alive` : `dead`}</em></Div>
      {render && <Component {...(props || {})} />}
    </Div>
  )
}
