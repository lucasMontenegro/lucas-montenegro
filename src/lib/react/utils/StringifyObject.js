import stringifyObject from "stringify-object"
import React from "react"
import Div from "lib/react/utils/Div"
import PropTypes from "prop-types"
export default function StringifyObject ({ id, className, source }) {
  const text = stringifyObject(source, {
    indent: `  `,
    singleQuotes: false,
  })
  return <Div color="RosyBrown"><pre id={id} className={className}>{text}</pre></Div>
}
StringifyObject.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
}