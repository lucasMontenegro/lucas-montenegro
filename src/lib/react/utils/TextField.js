import React, { useState } from "react"
import Div from "lib/react/utils/Div"
import MuiTextField from "@material-ui/core/TextField"
import Button from "@material-ui/core/Button"
const style = {
  display: `flex`,
  alignItems: `flex-end`,
}
export default function TextField (props) {
  const [value, saveValue] = useState(props.initialValue)
  const id = props.id || `text-field`
  return (
    <Div id={id} style={style} color="PaleVioletRed">
      <Div>
        <MuiTextField
          className="field"
          label={id}
          type="text"
          size="40"
          value={value}
          onChange={e => saveValue(e.target.value)}
        />
      </Div>
      <Div>
        <Button className="button" variant="outlined" onClick={() => props.onSave(value)}>
          SAVE
        </Button>
      </Div>
    </Div>
  )
}