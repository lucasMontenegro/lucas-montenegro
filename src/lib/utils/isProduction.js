import globals from "lib/utils/globals"
export default function isProduction () {
  return (
    globals.process ? globals.process.env.NODE_ENV === `production` :
    process.env.NODE_ENV === `production`
    // WebPack will replace this last line with a boolean
  )
}