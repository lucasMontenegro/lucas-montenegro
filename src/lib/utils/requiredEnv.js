import globals from "lib/utils/globals"
export default function requiredEnv (key) {
  const { env } = globals.process
  if (key in env) {
    return env[key]
  }
  throw Error(`process.env.${key} is required`)
}