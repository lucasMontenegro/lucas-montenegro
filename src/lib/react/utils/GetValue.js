import React from "react"
import Div from "lib/react/utils/Div"
import Button from "@material-ui/core/Button"
import StringifyObject from "lib/react/utils/StringifyObject"
import PropTypes from "prop-types"
export default function GetValue ({ id, className, value, onClick }) {
  return (
    <Div id={id} className={className} color="black">
      <h5>{id || className}</h5>
      <Div>
        <Button
          className="button"
          variant="outlined"
          onClick={onClick}
        >
          GET VALUE
        </Button>
      </Div>
      <StringifyObject className="value" source={value} />
    </Div>
  )
}
GetValue.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired,
}