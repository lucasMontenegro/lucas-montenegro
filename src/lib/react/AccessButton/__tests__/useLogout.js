jest.mock(`lib/utils/globals`, () => ({ __esModule: true, default: {} }))
jest.mock(`react`, () => ({ __esModule: true, useState: jest.fn() }))
describe(`../useLogout`, () => {
  jestUtils.describeDependencies({
    deps: [
      `lib/utils/globals`,
      `react`,
    ],
    relativeBasePath: __dirname,
  })
  {
    const auth0 = { logout: jest.fn() }
    function itShouldOpenTheDialog(setIsOpen) {
      it(`should open the dialog`, () => {
        expect(setIsOpen.mock.calls).toEqual([[true]])
        setIsOpen.mockClear()
      })
      it(`should not log out`, () => {
        expect(auth0.logout.mock.calls).toEqual([])
      })
    }
    function itShouldLogOut (setIsOpen) {
      it(`should not open the dialog`, () => {
        expect(setIsOpen.mock.calls).toEqual([])
      })
      it(`should log out`, () => {
        expect(auth0.logout.mock.calls).toEqual([[]])
        auth0.logout.mockClear()
      })
    }
    const cases = [
      [`BYPASS_DIALOG`, true, {
        initialAuto: false,
        testOpen: itShouldLogOut,
        newAuto: false,
        newStorageValue: `BYPASS_DIALOG`,
      }],
      [`OPEN_DIALOG`, true, {
        initialAuto: false,
        testOpen: itShouldOpenTheDialog,
        newAuto: false,
        newStorageValue: `BYPASS_DIALOG`,
      }],
      [null, true, {
        initialAuto: true,
        testOpen: itShouldOpenTheDialog,
        newAuto: false,
        newStorageValue: `BYPASS_DIALOG`,
      }],
      [`BYPASS_DIALOG`, false, {
        initialAuto: false,
        testOpen: itShouldLogOut,
        newAuto: true,
        newStorageValue: `OPEN_DIALOG`,
      }],
      [`OPEN_DIALOG`, false, {
        initialAuto: false,
        testOpen: itShouldOpenTheDialog,
        newAuto: true,
        newStorageValue: `OPEN_DIALOG`,
      }],
      [null, false, {
        initialAuto: true,
        testOpen: itShouldOpenTheDialog,
        newAuto: true,
        newStorageValue: `OPEN_DIALOG`,
      }],
    ]
    const msg = `useLogout (storage value %j, auto %j)`
    describe.each(cases)(msg, (storageValue, auto, expected) => {
      const localStorage = { getItem: () => storageValue }
      let useState
      const setAuto = jest.fn()
      const setIsOpen = jest.fn()
      let logout
      beforeAll(() => {
        require("lib/utils/globals").default.window = { localStorage }
        ;({ useState } = require("react"))
        useState.mockReturnValueOnce([auto, setAuto])
        useState.mockReturnValueOnce([false, setIsOpen])
        logout = require("../useLogout").default(auth0)
      })
      afterAll(() => {
        jest.resetModules()
      })
      it(`should set up the logout state`, () => {
        expect(logout).toMatchSnapshot()
      })
      it(`should set up the initial value of logout.auto`, () => {
        expect(useState.mock.calls).toEqual([[expected.initialAuto], [false]])
        useState.mockClear()
      })
      describe(`logout.open`, () => {
        beforeAll(() => {
          logout.open()
        })
        expected.testOpen(setIsOpen)
      })
      describe(`logout.close`, () => {
        it(`should close the dialog`, () => {
          logout.close()
          expect(setIsOpen.mock.calls).toEqual([[false]])
          setIsOpen.mockClear()
        })
      })
      describe(`logout.handleAuto`, () => {
        it(`should update the checkbox state`, () => {
          logout.handleAuto()
          expect(setAuto.mock.calls).toEqual([[expected.newAuto]])
          setAuto.mockClear()
        })
      })
      describe(`logout.confirm`, () => {
        const setItem = jest.fn()
        beforeAll(() => {
          localStorage.setItem = setItem
          logout.confirm()
          delete localStorage.setItem
        })
        it(`should update the storage`, () => {
          expect(setItem.mock.calls).toEqual([[
            `lib-react-access_button-use_logout`,
            expected.newStorageValue,
          ]])
          setItem.mockClear()
        })
        it(`should logout`, () => {
          expect(auth0.logout.mock.calls).toEqual([[]])
          auth0.logout.mockClear()
        })
      })
    })
  }
})