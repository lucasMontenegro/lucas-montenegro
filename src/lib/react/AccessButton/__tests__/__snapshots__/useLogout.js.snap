// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`../useLogout external dependencies react should match the installed version 1`] = `"^16.10.1"`;

exports[`../useLogout local dependencies lib/utils/globals should match the saved hash 1`] = `"658dffcc9a7d7107301d9ee254ba487ada2e2497635e046c509db0828ff3533f"`;

exports[`../useLogout useLogout (storage value "BYPASS_DIALOG", auto false) should set up the logout state 1`] = `
Object {
  "auto": false,
  "close": [Function],
  "confirm": [Function],
  "handleAuto": [Function],
  "isOpen": false,
  "open": [Function],
}
`;

exports[`../useLogout useLogout (storage value "BYPASS_DIALOG", auto true) should set up the logout state 1`] = `
Object {
  "auto": true,
  "close": [Function],
  "confirm": [Function],
  "handleAuto": [Function],
  "isOpen": false,
  "open": [Function],
}
`;

exports[`../useLogout useLogout (storage value "OPEN_DIALOG", auto false) should set up the logout state 1`] = `
Object {
  "auto": false,
  "close": [Function],
  "confirm": [Function],
  "handleAuto": [Function],
  "isOpen": false,
  "open": [Function],
}
`;

exports[`../useLogout useLogout (storage value "OPEN_DIALOG", auto true) should set up the logout state 1`] = `
Object {
  "auto": true,
  "close": [Function],
  "confirm": [Function],
  "handleAuto": [Function],
  "isOpen": false,
  "open": [Function],
}
`;

exports[`../useLogout useLogout (storage value null, auto false) should set up the logout state 1`] = `
Object {
  "auto": false,
  "close": [Function],
  "confirm": [Function],
  "handleAuto": [Function],
  "isOpen": false,
  "open": [Function],
}
`;

exports[`../useLogout useLogout (storage value null, auto true) should set up the logout state 1`] = `
Object {
  "auto": true,
  "close": [Function],
  "confirm": [Function],
  "handleAuto": [Function],
  "isOpen": false,
  "open": [Function],
}
`;
